### PR TITLE
feat(tier4_vehicle_rviz_plugin): add acceleration visualization plugin to RViz

### PR DESCRIPTION
**Pull request by Owen-Liuyuxuan**
## Description

Add an acceleration meter plugin for debugging on RVIZ.

## Tests performed

PSimulation and Rosbag simulation.

## Interface changes

The new plugin relies on topics like /localization/acceleration, and expose plugin parameters as described in changes in README.  

## Effects on system behavior

A new panel recording acceleration is expected to opened on the right of speed panel.
Screenshot from 2023-08-04 13-48-00

The center text should turn red (by default) if acceleration is larger than a max-threshold (re-configurable on Rviz), or smaller than a min-threshold. (In the screenshot we manually reconfigured the thresholds to demonstrate the behaviour)
Screenshot from 2023-08-04 13-48-45

 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by yukkysaito — CHANGES_REQUESTED** [review]
Thank you for your PR :+1: 

My Feedback: 
Is it possible to remove the limit part from this plugin as I would like to create it as a separate plugin?
https://github.com/autowarefoundation/autoware.universe/pull/4506/files#diff-b668fdfacd30844d8af0cdcaee2742a5c251f497dbf82ebbbd2793d88a9aceb1R182

Since the limiting acceleration is specified by a parameter, it is better to indicate it.

**Comment by Owen-Liuyuxuan** [comment]
> Thank you for your PR 👍
> 
> My Feedback: Is it possible to remove the limit part from this plugin as I would like to create it as a separate plugin? https://github.com/autowarefoundation/autoware.universe/pull/4506/files#diff-b668fdfacd30844d8af0cdcaee2742a5c251f497dbf82ebbbd2793d88a9aceb1R182
> 
> Since the limiting acceleration is specified by a parameter, it is better to indicate it.

Agree, it is not necessary to have a limit here. I will try update these parts.

**Comment by shmpwk** [comment]
@Owen-Liuyuxuan 
Could you check spell check CI ?

**Comment by shmpwk** [comment]
@Owen-Liuyuxuan 
You can ignore codecov fail for now.

**Comment by shmpwk** [comment]
@Owen-Liuyuxuan 

Looks like even if the acceleration is negative value, it show right side in the meter.
image

File: common/tier4_vehicle_rviz_plugin/src/tools/acceleration_meter.cpp [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+  QColor label_color(property_normal_text_color_->getColor());
+  text_color.setAlpha(255);
+  painter.setPen(QPen(label_color, static_cast<int>(2), Qt::SolidLine));
+  QFont label_font = painter.font();
+  font.setPixelSize(
+    std::max(static_cast<int>(static_cast<double>(w) * property_label_scale_->getFloat()), 1));
+  font.setBold(true);
+  painter.setFont(font);  // Use the font as the meter panel but smaller.
+  std::ostringstream label_ss;
+  label_ss << std::fixed << "Acceleration Meter";
```
Comment (shmpwk) [inline common/tier4_vehicle_rviz_plugin/src/tools/acceleration_meter.cpp:187]: IMO, "Acceleration Meter" is not needed for simplicity like other meters

Comment (Owen-Liuyuxuan) [reply common/tier4_vehicle_rviz_plugin/src/tools/acceleration_meter.cpp]: Agree. And sorry I should have checked more carefully for the meter hand computation. These two issues were refined in new pushes.

Comment (shmpwk) [reply common/tier4_vehicle_rviz_plugin/src/tools/acceleration_meter.cpp]: Thank you ! 
(It is ok to contribute tomorrow as a partime job. :smile:  )

File: common/tier4_vehicle_rviz_plugin/src/tools/acceleration_meter.hpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  rviz_common::properties::IntProperty * property_top_;
+  rviz_common::properties::IntProperty * property_length_;
+  rviz_common::properties::IntProperty * property_value_height_offset_;
+  rviz_common::properties::FloatProperty * property_value_scale_;
+  rviz_common::properties::FloatProperty * property_emergency_threshold_;
+  // QImage hud_;
+
+private:
+  static constexpr float meter_min_acceleration_ = -5.0f;
+  static constexpr float meter_max_acceleration_ = 5.0f;
```
Comment (shmpwk) [inline common/tier4_vehicle_rviz_plugin/src/tools/acceleration_meter.hpp:68]: The maximum and minimum limit of the acceleration is defined in the planning validator as 9.8 m/ss.

https://github.com/autowarefoundation/autoware_launch/blob/32d537b692b24549fa0d45a42ff6aa6e74c7a447/autoware_launch/config/planning/scenario_planning/common/planning_validator/planning_validator.param.yaml#L23-L24

Could you expand the parameters like

```suggestion
  static constexpr float meter_min_acceleration_ = -10.0f;
  static constexpr float meter_max_acceleration_ = 10.0f;
```

File: common/tier4_vehicle_rviz_plugin/README.md [hunk 3]
```diff
@@ -64,6 +65,19 @@ This plugin provides a visual and easy-to-understand display of vehicle speed, t
...
+| Name                             | Type   | Default Value        | Description                                  |
+| -------------------------------- | ------ | -------------------- | -------------------------------------------- |
+| `property_normal_text_color_`    | QColor | QColor(25, 255, 240) | Normal text color                            |
+| `property_emergency_text_color_` | QColor | QColor(255, 80, 80)  | Emergency acceleration color                 |
+| `property_left_`                 | int    | 896                  | Left of the plotter window [px]              |
+| `property_top_`                  | int    | 128                  | Top of the plotter window [px]               |
+| `property_length_`               | int    | 256                  | Height of the plotter window [px]            |
+| `property_value_height_offset_`  | int    | 0                    | Height offset of the plotter window [px]     |
+| `property_value_scale_`          | float  | 1 / 6.667            | Value text scale                             |
+| `property_emergency_threshold_`  | float  | 2.0                  | Acceleration threshold for emergency [m/s^2] |
```
Comment (shmpwk) [inline common/tier4_vehicle_rviz_plugin/README.md:79]: The ideas of emergency threshold is nice ! 

The common planning acceleration limits are defined here.
https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/planning/scenario_planning/common/common.param.yaml#L12-L13

```
      min_acc: -2.5         # min deceleration limit [m/ss]
      max_acc: 1.0          # max acceleration limit [m/ss]
```

We are more likely to set `min_acc` and `max_acc` as a different value.
I recommend you to seperate those threhold.

**Comment by shmpwk** [comment]
@Owen-Liuyuxuan 
Thank you for your fixing.
Could you update the PR description screen shot ?

**Review by shmpwk — APPROVED** [review]
LGTM! Wating for @yukkysaito approval.

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**